### PR TITLE
Replace "fs-extra" with hh utils methods

### PR DESCRIPTION
**Pull request by ChristopherDedominici**
- [ ] Because this PR includes a **bug fix**, relevant tests have been included.
- [x] Because this PR includes a **new feature**, the change was previously discussed on an Issue or with someone from the team.
- [ ] I didn't do anything of this.

---

### CODE REVIEW

File: v-next/hardhat-build-system/src/internal/utils/download.ts [hunk 1]
```diff
@@ -19,48 +21,27 @@ function resolveTempFileName(filePath: string): string {
...
-  extraHeaders: { [name: string]: string } = {},
-) {
-  const { getGlobalDispatcher, ProxyAgent, request } = await import("undici");
-
-  let dispatcher: Dispatcher;
-  if (process.env.http_proxy !== undefined && shouldUseProxy(url)) {
-    dispatcher = new ProxyAgent(process.env.http_proxy);
-  } else {
-    dispatcher = getGlobalDispatcher();
+export async function download(url: string, filePath: string) {
```
Comment (ChristopherDedominici) [inline v-next/hardhat-build-system/src/internal/utils/download.ts:24]: @kanej @alcuadrado @schaable check this function with caution, it downloads the compiler. Tests are passing, but it's worth spend some time on it

File: v-next/hardhat-build-system/src/internal/solidity/compiler/downloader.ts [hunk 2]
```diff
@@ -293,12 +302,12 @@ export class CompilerDownloader implements ICompilerDownloader {
...
   async #shouldDownloadCompilerList(): Promise<boolean> {
     const listPath = this.#getCompilerListPath();
-    if (!(await fsExtra.pathExists(listPath))) {
+    if (!(await exists(listPath))) {
       return true;
     }
 
-    const stats = await fsExtra.stat(listPath);
-    const age = new Date().valueOf() - stats.ctimeMs;
+    const ctime = await getChangeTime(listPath);
```
Comment (ChristopherDedominici) [inline v-next/hardhat-build-system/src/internal/solidity/compiler/downloader.ts:309]: @kanej @schaable spend a few more seconds here just to double check. This change is a bit more "complex"

File: v-next/hardhat-build-system/src/internal/types/artifacts.ts [hunk 3]
```diff
@@ -41,11 +36,6 @@ export interface Artifacts {
    */
   getBuildInfo(fullyQualifiedName: string): Promise<BuildInfo | undefined>;
 
-  /**
-   * Synchronous version of getBuildInfo.
-   */
-  getBuildInfoSync(fullyQualifiedName: string): BuildInfo | undefined;
```
Comment (kanej) [inline v-next/hardhat-build-system/src/internal/types/artifacts.ts:47]: This is because there are no usages from the build system?

Comment (kanej) [reply v-next/hardhat-build-system/src/internal/types/artifacts.ts:47]: Or is it because utils implies async to you are removing non-async versions?

**Review by kanej — APPROVED** [review]
Looks good.

File: v-next/hardhat-build-system/src/internal/types/artifacts.ts (hunk 3 above)
Comment (ChristopherDedominici) [reply v-next/hardhat-build-system/src/internal/types/artifacts.ts:47]: Yes, correct

**Pull request merged**